### DISCUSSION TITLE
async attribute converters

### DISCUSSION THREAD

**Discussion by pragmasoft-ua** [lit/lit#3687] [Ideas]
I propose to enhance attribute converters to support async methods. 

One use case I have is having a properly of type File with an attribute src accepting URL. To convert URL to file with a fetch, I need an async converter support.

**Comment by justinfagnani** [comment]
I think [@lit-labs/task](https://www.npmjs.com/package/@lit-labs/task) is a much better way to handle this. There are all kinds of considerations, like failure states and race conditions, that Task already takes care of. I'm working on lit.dev docs for Task right now.

**Reply by pragmasoft-ua** [reply]
Thanks for pointing at it, will have a look
